File: src/main/java/com/itmo/microservices/tasks/entity/Task.java
package com.itmo.microservices.tasks.entity;

import com.itmo.microservices.tasks.model.TaskStatus;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.Id;
import java.util.UUID;

@Entity
@Data
@AllArgsConstructor
@NoArgsConstructor
@Builder
public class Task {

    @Id
    @GeneratedValue
    private UUID id;
    private String author;
    private String assignee;
    private String description;
    private TaskStatus status;

    public void start() {
        this.status = TaskStatus.IN_PROCESS;
    }

    public void complete() {
        this.status = TaskStatus.DONE;
    }
}

File: src/main/kotlin/com/itmo/microservices/demo/tasks/impl/entity/Task.kt
package com.itmo.microservices.demo.tasks.impl.entity

import com.itmo.microservices.demo.tasks.api.model.TaskStatusEnum
import java.util.*
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.Id

@Entity
class Task {

    @Id
    @GeneratedValue
    var id: UUID? = null
    var author: String? = null
    var assignee: String? = null
    var title: String? = null
    var description: String? = null
    var status: TaskStatusEnum = TaskStatusEnum.TODO

    constructor()

    constructor(id: UUID? = null, author: String? = null, assignee: String? = null, title: String, description: String? = null, status: TaskStatusEnum) {
        this.id = id
        this.author = author
        this.assignee = assignee
        this.title = title
        this.description = description
        this.status = status
    }

    override fun toString(): String =
            "Task(id=$id, author=$author, assignee=$assignee, title=$title, description=$description, status=$status)"

}
